src/csvcubed/utils/sparql_handler/data_cube_state.py

```diff
@@ -8,38 +8,60 @@
 
 from dataclasses import dataclass
 from functools import cached_property
-from typing import Dict, List, Optional
+from typing import Dict, List, Optional, TypeVar
 
 from csvcubed.models.cube.cube_shape import CubeShape
 from csvcubed.models.sparqlresults import (
+    ColumnDefinition,
     CubeTableIdentifiers,
     IsPivotedShapeMeasureResult,
     QubeComponentsResult,
     UnitResult,
 )
 from csvcubed.utils.dict import get_from_dict_ensure_exists
 from csvcubed.utils.iterables import first, group_by
-from csvcubed.utils.sparql_handler.csvw_state import CsvWState
+from csvcubed.utils.sparql_handler.csvw_inspector import CsvWInspector
 from csvcubed.utils.sparql_handler.sparqlquerymanager import (
+    select_column_definitions,
     select_csvw_dsd_qube_components,
     select_data_set_dsd_and_csv_url,
     select_is_pivoted_shape_for_measures_in_data_set,
     select_units,
 )
 
+T = TypeVar("T")
+
 
 @dataclass
 class DataCubeState:
-    """Provides access to inspect the data cubes contained in an rdflib graph."""
+    csvw_inspector: CsvWInspector
+
+    """
+    Private utility functions.
+    """
+
+    def _get_value_for_key(self, key: str, dict: Dict[str, T]) -> T:
```

Comment:
Please don't use this. Use the function we have in `csvcubed.utils.dict` (`get_from_dict_ensure_exists`) instead since it does the same thing.

We deliberately removed this function from this class.

Reply:
Removed